subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/AbstractNestedRuntimeBeanNode.java

```diff
@@ -39,26 +39,26 @@
 import java.util.Queue;
 
 public abstract class AbstractNestedRuntimeBeanNode extends RuntimeBeanNode<Object> {
+    private final TypeMetadata typeMetadata;
+
     protected AbstractNestedRuntimeBeanNode(@Nullable RuntimeBeanNode<?> parentNode, @Nullable String propertyName, Object bean, TypeMetadata typeMetadata) {
-        super(parentNode, propertyName, bean, typeMetadata);
+        super(parentNode, propertyName, bean);
+        this.typeMetadata = typeMetadata;
     }
 
     protected void visitProperties(PropertyVisitor visitor, final Queue<RuntimeBeanNode<?>> queue, final RuntimeBeanNodeFactory nodeFactory, TypeValidationContext validationContext) {
-        TypeMetadata typeMetadata = getTypeMetadata();
         typeMetadata.visitValidationFailures(getPropertyName(), validationContext);
         for (PropertyMetadata propertyMetadata : typeMetadata.getPropertiesMetadata()) {
             PropertyAnnotationHandler annotationHandler = typeMetadata.getAnnotationHandlerFor(propertyMetadata);
-            if (annotationHandler.shouldVisit(visitor)) {
-                String propertyName = getQualifiedPropertyName(propertyMetadata.getPropertyName());
-                PropertyValue value = new BeanPropertyValue(getBean(), propertyMetadata.getGetterMethod());
-                annotationHandler.visitPropertyValue(
-                    propertyName,
-                    value,
-                    propertyMetadata,
-                    visitor,
-                    (childPropertyName, bean) -> queue.add(nodeFactory.create(AbstractNestedRuntimeBeanNode.this, childPropertyName, bean))
-                );
-            }
+            String propertyName = getQualifiedPropertyName(propertyMetadata.getPropertyName());
```

Comment:
Is `annotationHandler.shouldVisit(visitor)` not needed anymore?

Reply:
This is intentional, and yes, it was unnecessary.